source/backend/dal/Repositories/DocumentQueueRepository.cs

```diff
@@ -32,6 +34,36 @@ public DocumentQueueRepository(
 
         #region Methods
 
+        /// <summary>
+        /// Add Document to Queue.
+        /// </summary>
+        /// <param name="queuedDocument"></param>
+        /// <returns></returns>
+        public PimsDocumentQueue Add(PimsDocumentQueue queuedDocument)
+        {
+            queuedDocument.ThrowIfNull(nameof(queuedDocument));
+
+            // Default values for new queue items.
+            queuedDocument.DocumentQueueStatusTypeCode = DocumentQueueStatusTypes.PENDING.ToString();
+            queuedDocument.DataSourceTypeCode = DataSourceTypes.PIMS.ToString();
+            queuedDocument.MayanError = null;
+
+            // Add
+            Context.PimsDocumentQueues.Add(queuedDocument);
+
+            return queuedDocument;
+        }
+
+        /// <summary>
+        /// Find Queue Item for a Document.
+        /// </summary>
+        /// <param name="documentId"></param>
+        /// <returns></returns>
+        public PimsDocumentQueue GetByDocumentId(long documentId)
+        {
+            return Context.PimsDocumentQueues.Where(x => x.DocumentId == documentId).FirstOrDefault();
```

Comment:
should this be asNoTracking?

Reply:
Needed to update the mayanId to NULL after successful delete call to Mayan API.